### PR TITLE
[vulkan] Add core graph components

### PR DESCRIPTION
**Pull request by SS-JIA**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #92994

Differential Revision: [D42614155](https://our.internmc.facebook.com/intern/diff/D42614155/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D42614155/)!

### CODE REVIEW

File: aten/src/ATen/native/vulkan/graph/Value.h [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      int64_t as_int;
+      double as_double;
+      bool as_bool;
+    } u;
+
+    vTensor as_tensor;
+    TensorStaging as_staging;
+
+    Payload() : u() {}
+    ~Payload() {}
```
Comment (Skylion007) [inline aten/src/ATen/native/vulkan/graph/Value.h:42]:
```suggestion
    ~Payload() = default;
```

File: aten/src/ATen/native/vulkan/api/Tensor.cpp [hunk 2]
```diff
@@ -266,6 +268,47 @@ vTensor::vTensor(
...
+      memory_format_{other.memory_format_},
+      sizes_(std::move(other.sizes_)),
+      strides_(std::move(other.strides_)),
+      gpu_sizes_(std::move(other.gpu_sizes_)),
+      gpu_strides_(std::move(other.gpu_strides_)),
+      metadata_uniform_(std::move(other.metadata_uniform_)),
+      is_quantized_{other.is_quantized_},
+      q_scale_{other.q_scale_},
+      q_zero_point_{other.q_zero_point_},
+      view_(std::move(other.view_)) {}
```
Comment (Skylion007) [inline aten/src/ATen/native/vulkan/api/Tensor.cpp:310]: Why not add move assignment for completeness?

File: aten/src/ATen/native/vulkan/api/Tensor.cpp [hunk 3]
```diff
@@ -2,6 +2,8 @@
 #include <ATen/native/vulkan/api/Utils.h>
 #include <c10/util/accumulate.h>
 
+#include <iostream>
```
Comment (Skylion007) [inline aten/src/ATen/native/vulkan/api/Tensor.cpp:5]: Do you still need this include?

**Review by salilsdesai — APPROVED** [review]
Lgtm

File: aten/src/ATen/native/vulkan/graph/Value.h (hunk 1 above)
Comment (SS-JIA) [reply aten/src/ATen/native/vulkan/graph/Value.h:42]: good point, adding that as well.

File: aten/src/ATen/native/vulkan/api/Tensor.cpp (hunk 2 above)
Comment (SS-JIA) [reply aten/src/ATen/native/vulkan/api/Tensor.cpp]: ended up adding a defaulted move assignment as well

File: aten/src/ATen/native/vulkan/api/Tensor.cpp (hunk 3 above)
Comment (SS-JIA) [reply aten/src/ATen/native/vulkan/api/Tensor.cpp]: good catch, removed

**Comment by SS-JIA** [comment]
Unable to sync with Phabricator diff anymore via `ghexport`. Will close this PR and export it again into another PR.

**Comment by SS-JIA** [comment]
New PR at: https://github.com/pytorch/pytorch/pull/94221

**Pull request closed without merging**